cogs/ranked.py

```diff
@@ -1585,7 +1585,7 @@ async def edit_match(self, interaction: discord.Interaction, player: discord.Mem
 
     
     @app_commands.command(description="Submit Score")
-    @app_commands.checks.cooldown(1, 20.0, key=lambda i: i.guild_id)
+    @app_commands.checks.cooldown(1, 120.0, key=lambda i: (i.guild_id,i.current_match.id))
```

Comment:
`current_match` doesn't exist on `i` (interaction) but can be looked up with the function (see below)
`id` doesn't exit on `current_ match` but we can create it by modifying the XrcGame class